app/models/cocktail.rb

```diff
@@ -2,10 +2,11 @@ class Cocktail < ApplicationRecord
   validates :name, :thumbnail, presence: true
 
   has_many :ratings, dependent: :destroy
-  def self.dashboard_five
-    select('*')
-    .from('cocktails')
-    .order('Random()')
-    .limit(5)
+  def self.dashboard_five(user_id)
+    left_outer_joins(:ratings)
+      .select('cocktails.*, case when ratings.stars is null then 0 else ratings.stars end as rating')
+      .where('ratings.user_id is null or ratings.user_id = ?', user_id)
+      .order(Arel.sql('RANDOM()'))
```

Comment:
Makes sense adding the user id for the search, I was curious if this would be used with a current user